### PR TITLE
"User: design discussion" abstract

### PR DESCRIPTION
**Pull request by msvrtan**
Please help me abstract this talk: I would love to hear suggestions on the abstract/title but also would appreciate feedback on how interesting you feel the subject is.

I'm preparing a set of new talks for 2019, I'm aware I dont have the time to prepare all of them but as they are 'my babies' it's rather hard to be objective on how interesting a particular topic is.

### CODE REVIEW

File: talks/user-design-discussion.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+As an example of software architecture most of us are familiar with: `Users`, lets look at some patterns and pitfalls of them. There is no right answer, as it does depend on your particular use case but idea behind this session is to expand our knowledge beyond 'your mileage may vary' and 'it depends' that have become so often a generic answer to so many questions.
+
+This session is not meant as a talk but as a discussion: it's curcial that you as attendee get involved and help in the discussion: whether it's you asking questions, offering ideas or your use cases and solutions. Initial examples will be prepared by me and I will moderate the discussion to keep it on track and avoid some needless bikeshedding ;)
+
+Key takeaway from this session: share your experiences and learn from other people!
+
+
+### Additional information:
+
+I would love to do this discussion on ZgPHP in May or June of 2019, as 2nd in the series (for now I have only 2 topics for sessions).
```
Comment (zoka123) [inline talks/user-design-discussion.md]: I'd be happy to hear thoughts on:
- naming (users, clients, members, subscribers, visitors, atendees ...)
- separating authentication logic from domain logic (auth user vs app user and his profile)
- "userless" and stateless authentication and authorization implementation (e.g. for machine to machine authentication, OAuth2)
- Outsourcing authentication (E.g. social login or Firebase auth)

I'd also like to discuss Smyfony auth approach that is IMO too much bound to the "user concept" 

(Look at method names such as `loadByUsername`, and `getUser`, UserInterface required and so on https://symfony.com/doc/current/security/user_provider.html#custom-user-provider, https://symfony.com/doc/current/security/guard_authentication.html)

Comment (pelshoff) [inline talks/user-design-discussion.md]: Very interesting topic; perhaps you could work together with upcoming conferences to do it as a panel discussion like GrowIT?

Comment (msvrtan) [reply talks/user-design-discussion.md]: I will keep it on my mind for a panel-like discussion!

For now, I would like to first run it as a discussion and than go forward: I'm not sure that a panel discussion in it's normal format would be suitable without people being able to maybe 'code' a short example or at least show structure they are referring to.

Comment (msvrtan) [reply talks/user-design-discussion.md]: Thanks, I'we added them to notes so I dont forget it!

**Review by HcsOmot — COMMENTED** [review]
I can't find the original source and author, so I'm explicitly stating that I've read this somewhere random online:
"Only drug dealers and software developers have 'users'".

This might be a nice tie-in with the proposal #5 - I think one of the issues here is the naming problem, but perhaps a smaller one in comparison to the often overloaded concerns of a 'user' in a system.

I'd love to hear more on the topics @zoka123 suggested, but would also love to learn more about properly separating various types of 'users' within the system - when to design for different 'user' types and how to do it properly.

I haven't been able to find some good resources online that deal with this subject and haven't seen any conference talks on this subject either.

Almost all online available learning materials (blog posts, video tutorials, various articles etc) only ever mention one type of users in the system, giving various accounts different access roles, but essentially treating them all the same.

I believe the concept of introducing different types of users comes from DDD world, so I'd love to hear more about that as well.

**Comment by msvrtan** [comment]
Well the idea is to go thru some 'basic' examples and discuss how different they were from people who attended: like we all think user has to have a username and password while internet banking uses neither :)

File: talks/user-design-discussion.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+### Title:
+User: design discussion
+
+### Abstract:
+
+As an example of software architecture most of us are familiar with: `Users`, lets look at some patterns and pitfalls of them. There is no right answer, as it does depend on your particular use case but idea behind this session is to expand our knowledge beyond 'your mileage may vary' and 'it depends' that have become so often a generic answer to so many questions.
```
Comment (DamirSvrtan) [inline talks/user-design-discussion.md]: Have to say that I had to really concentrate on understanding what is going on here. First I got thrown away by the title as I had a feeling like it's a user experience type of talk.

Maybe a talk subject could be "Designing the User Entity".

Maybe I'd start with something like:

_Every app has some concept of a **User**, but is it okay to cram up all the info into one entity, or is it better to separate it in their domain entities such as members, employees, visitors? There is no right answer, as it does depend on your particular use case! Let's go above the 'it depends' and 'your milage may vary', and look at some real world use cases!_

File: talks/user-design-discussion.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+This session is not meant as a talk but as a discussion: it's curcial that you as attendee get involved and help in the discussion: whether it's you asking questions, offering ideas or your use cases and solutions. Initial examples will be prepared by me and I will moderate the discussion to keep it on track and avoid some needless bikeshedding ;)
```
Comment (DamirSvrtan) [inline talks/user-design-discussion.md]: I see that most of the talks have this "let's make it a discussion" part of the abstract, I'd just make it a little bit more motivating. I wrote something like this in the previous talk abstract:

_Let's not make this talk a one way street - I'd love that you as attendee participate and share your stories with all of us, as I will leave some time at the end of the talk for a short discussion._

**Pull request merged**